pkg/argoapp/argoapp.go

```diff
@@ -117,3 +120,30 @@ func NewApplication(config ApplicationConfig) (*unstructured.Unstructured, error
 
 	return &unstructured.Unstructured{Object: obj}, nil
 }
+
+func NewApplication(config ApplicationConfig) (*v1alpha1.Application, error) {
```

Comment:
That's a breaking change, but I guess it's ok because we (biscuit) are the only user of the code.

Reply:
I can update `architect` as needed.